experiment_prototype/scan_classes/sequences.py

```diff
@@ -256,38 +321,70 @@ def make_pulse_dict(pulse_timing_info):
         self.numberofreceivesamples = int(self.transmit_metadata['rx_sample_rate'] * self.sstime *
                                           1e-6)
 
-        sample_time = (int(self.combined_pulses_metadata[0]['total_pulse_len'] / 2) + tr_window_time) / txrate
-        self.first_rx_sample_time = sample_time
+        self.output_encodings = collections.defaultdict(list)
+
+
+        # create debug dict for tx samples.
+        debug_dict = {'txrate' : txrate,
+                      'txctrfreq' : txctrfreq,
+                      'pulse_timing' : [],
+                      'pulse_sample_start' : [],
+                      'sequence_samples' : {},
+                      'decimated_samples' : {},
+                      'dmrate' : dm_rate
+                      }
+
+
+        for i, cpm in enumerate(combined_pulses_metadata):
+            debug_dict['pulse_timing'].append(cpm['start_time_us'])
+            debug_dict['pulse_sample_start'].append(cpm['pulse_sample_start'])
+
+        for i in range(main_antenna_count):
+            debug_dict['sequence_samples'][i] = []
+            debug_dict['decimated_samples'][i] = []
+
+        self.debug_dict = debug_dict
+
+        first_slice_pulse_len = self.combined_pulses_metadata[0]['component_info'][0]['pulse_num_samps']
+        full_pulse_samps = first_slice_pulse_len + 2 * tr_window_num_samps
+        offset_to_start = int(full_pulse_samps/2)
+        self.first_rx_sample_start = offset_to_start
 
         self.blanks = self.find_blanks()
 
-        self.output_encodings = collections.defaultdict(list)
+
 
     def make_sequence(self, beam_iter, sequence_num):
         """
         Create the samples needed for each pulse in the sequence. This function is optimized to
         be able to generate new samples every sequence if needed.
+        Modifies the samples_array and isarepeat fields of all pulse
+        dictionaries needed for this sequence for
+        radar_control to use in operation.
```

Comment:
Sorry - I found myself looking into how this works again so thought I'd suggest more to the docstring here; just a suggestion

```suggestion
        radar_control to use in operation. 
        
        The beam steering phases for the samples are applied in the sequence init; this function will take 
        those samples, finalize the timing of the pulses, and apply additional phase encodings if necessary.
```